assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -154,14 +169,92 @@ const baseActions = {
 
 		return { response, error };
 	},
+
+	/**
+	 * Registers a module
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string}          slug                         Module slug.
+	 * @param {Object}          settings                     Module settings.
+	 * @param {string}          [settings.name]              Optional. Module name. Default is the slug.
+	 * @param {string}          [settings.description]       Optional. Module description. Default empty string.
+	 * @param {string}          [settings.icon]              Optional. Module icon. Default empty string.
+	 * @param {number}          [settings.order]             Optional. Numeric indicator for module order. Default 10.
+	 * @param {string}          [settings.homepage]          Optional. Module homepage URL. Default empty string.
+	 * @param {React.Component} [settings.settingsComponent] React component to render the settings panel. Default is the DefaultModuleSettings component.
+	 * @return {Object} Redux-style action.
+	 */
+	*registerModule( slug, { settingsComponent = DefaultModuleSettings, ...settings } = {} ) {
+		invariant( slug, 'module slug is required' );
+
+		const registry = yield commonActions.getRegistry();
+		let registryKey = yield registry.select( CORE_SITE ).getRegistryKey();
+		if ( registryKey === undefined ) {
+			registryKey = uuidv4();
```

Comment:
So, as @eugene-manuilov mentioned: symbols would give us UUIDs. But then I thought about it more after the call and they won't properly serialise to POJOs… take, for instance:

```js
const a = {};
const b = Symbol('key');
const c = Symbol('key');
a[b] = 1;
a[c] = 2;
print(a);
> { [Symbol(key)]: 1, [Symbol(key)]: 2 }
JSON.stringify(a)
> '{}'
```

Symbols aren't as good a fit for redux as using strings, so let's stick with the UUID approach.

Reply:
How much does the `uuid` package increase bundle size? If there is a significant impact, we should reconsider something else - in real world usage the symbol approach would still work reliably I think?